jgrapht-core/src/main/java/org/jgrapht/alg/partition/BipartitePartitionFinder.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * (C) Copyright 2016-2018, by Dimitrios Michail, Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.partition;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.PartitionAlgorithm;
+
+import java.util.*;
+
+import static org.jgrapht.GraphTests.isEmpty;
+
+/**
+ * Algorithm for computing bipartite partitions thus checking whether a graph is bipartite or not.
+ *
+ * <p>
+ * The algorithm runs in linear time in the number of vertices and edges.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ * @author Alexandru Valeanu
+ * @since June 2018
+ */
+public class BipartitePartitionFinder<V, E> implements PartitionAlgorithm<V> {
```

Comment:
Should we just name this thing "BipartitePartitioning<V,E> implements PartitioningAlgorithm" instead of "BipartitePartitionFinder?